core/src/main/java/nl/nn/adapterframework/util/AppConstants.java

```diff
@@ -0,0 +1,232 @@
+/*
+   Copyright 2013, 2016 - 2019 Nationale-Nederlanden, 2020 - 2023 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.util;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Properties;
+import java.util.StringTokenizer;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.annotation.Nonnull;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Singleton class that has the constant values for this application. <br/>
+ * <p>When an instance is created, it tries to load the properties file specified
+ * by the <code>propertiesFileName</code> field</p>
+ * <p>If a property exits with the name <code>ADDITIONAL.PROPERTIES.FILE</code>
+ * that file is loaded also</p>
+
+ * @author Niels Meijer
+ * @version 2.1
+ *
+ */
+public final class AppConstants extends PropertyLoader {
+	private static final Logger LOG = LogManager.getLogger(AppConstants.class);
+
+	private static final String APP_CONSTANTS_PROPERTIES_FILE = "AppConstants.properties";
+	private static final String ADDITIONAL_PROPERTIES_FILE_KEY = "ADDITIONAL.PROPERTIES.FILE";
+	public static final String APPLICATION_SERVER_TYPE_PROPERTY = "application.server.type";
+	public static final String APPLICATION_SERVER_CUSTOMIZATION_PROPERTY = "application.server.type.custom";
+	public static final String ADDITIONAL_PROPERTIES_FILE_SUFFIX_KEY = ADDITIONAL_PROPERTIES_FILE_KEY+".SUFFIX"; //Can't be final because of tests
```

Comment:
I like the comment, because it is `final` already ;-)

Reply:
Not sure where this comment originally came from, but I've removed it :)